docs/guide-files/threading.md

```diff
@@ -0,0 +1,33 @@
+
+>NOTE: When a child protocol is a simple two-party interaction, mentioning the `pthid` in the first message of the child interaction is enough to establish context. However, in protocols involving more than two parties, the first message of the child protocol may not be seen by everyone, so simply mentioning `pthid` once may not provide enough context. Therefore, the rule is that each party in a child protocol MUST learn the identity of the parent thread via the first child protocol message they see. The simplest way to ensure this is to mention the `pthid` with every message in the child protocol. 
+
+Creating a connection is another classic interaction that lends itself to a parent-child paradigm. Alice and Bob rarely connect just to connect; usually they connect so they can conduct some type of business. In such situations, the connection protocol is a parent thread, and whatever business they then conduct becomes a child.
+
+Reporting and resolving errors and warnings is also a good use case for parent and child threads. It is true that problems can be communicated in specialized messages of an individual protocol; some protocols may make this choice. However, problems are often the gateway to deeper troubleshooting or even support tickets, which are complex workflows in their own right. Also, robust mechanisms for logging, analyzing, and reacting to problems may be reusable in many workflows. For this reason, it is recommended to model problem reporting as a child protocol, with an arbitrarily complex thread of messages as child to the parent workflow that triggered them. See [Reporting and Handling Problems](#reporting-and-handling-problems) for more info.
+
+Sometimes the parent of a thread is not known, or is so independent of its child as to be irrelevant. If Alice and Bob connected three years ago, each new interaction that they begin doesn't need to declare a `pthid` that references the original interaction that connected them. Declare a parent relationship for a thread when it is likely to be useful.
+
+### DIDComm Message URIs
+
+The `id`, `thid`, and `pthid` properties of any DIDComm message may be combined to form a URI that uniquely identifies the message (e.g., in debuggers, in log files, in archives). When this is further combined with [JSPath](https://github.com/dfilatov/jspath#quick-example), individual elements of messages may also be hyperlinked. See [Linkable Message Paths](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0217-linkable-message-paths/README.md). Such a scheme is out of scope for this spec, and support for it is NOT required of implementers. However, this spec reserves the `didcomm://` URI prefix for future work with semantics like these.
```

Comment:
I think most of this paragraph should remain here, but that semantic reservation probably needs to go back into the spec for greater awareness about the topic even if we may leave it as non-normative. Take a look at [this note](https://w3c.github.io/did-core/#did-url-syntax) in did-core about how the `;` charter was "reserved" without being normative.

Reply:
What would be the "semantic reservation" part that you'd move back?

Reply:
```suggestion
It's generally considered this functionality will be combined with [JSPath](https://github.com/dfilatov/jspath#quick-example), [JS Pointer](https://datatracker.ietf.org/doc/html/rfc6901), or individual elements of messages may also be hyperlinked. For a good idea of the concepts that we'd like to enable with this capability, see [Linkable Message Paths](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0217-linkable-message-paths/README.md).
```

Reply:
Okay, I took this out of the guide and put it back into the spec, with verbiage that tracks what you suggested here. You are attributed as author of the commit.